Makefile

```diff
@@ -22,14 +23,32 @@ gofmt:
 $(GOLINT):
 	go install golang.org/x/lint/golint
 
+$(GEN_ATOMICINT): $(wildcard ./internal/gen-atomicint/*)
+	go build -o $@ ./internal/gen-atomicint
+
 .PHONY: golint
 golint: $(GOLINT)
 	$(GOLINT) ./...
 
 .PHONY: lint
-lint: gofmt golint
+lint: gofmt golint generatenodirty
 
 .PHONY: cover
 cover:
 	go test -coverprofile=cover.out -coverpkg ./... -v ./...
 	go tool cover -html=cover.out -o cover.html
+
+.PHONY: generate
+generate: $(GEN_ATOMICINT)
+	go generate ./...
+
+.PHONY: generatenodirty
+generatenodirty:
+	@[ -z "$$(git status --porcelain)" ] || ( \
+		echo "Working tree is dirty. Commit your changes first."; \
+		exit 1 )
+	@make generate
+	@git diff --quiet || ( \
```

Comment:
This won't notice if a new file is added (while the `git status --porcelain` will above). Intentional?

Reply:
Good catch. Will switch to git status here and in the message as well.